test/game_test.rb

```diff
@@ -60,93 +60,159 @@ def test_it_has_all
   end
 
-  def test_average_goals_per_csv_header
-    goals = [10, 20]
-    season_lengths = [3, 8]
-    assert_equal [3.33, 2.5], Game.average_goals(goals, season_lengths)
-  end
 #deliverable
-  def test_average_goals_by_season
-    Game.all.stubs(:map).returns([20122013, 20162017]) #reconsider stubs map
-    Game.stubs(:average_goals).returns([3.33, 2.5])
+  def test_average_goals_by_season_and_divide_hash_values
+    stub_games_goals = {
+                        20122013 => {:goals => 10, :games_played => 3},
+                        20162017 => {:goals => 10, :games_played => 4}
+                        }
+    Game.stubs(:games_goals_by_season).returns(stub_games_goals)
     expected = {20122013 => 3.33, 20162017 => 2.5}
     assert_equal expected, Game.average_goals_by_season
+    assert_equal expected, Game.divide_hash_values(:goals, :games_played, Game.games_goals_by_season)
   end
 
-  def test_it_can_calculate_highest_total_score
-    assert_equal 5, @game.highest_total_score
-  end
-
-  def test_it_can_calculate_lowest_total_score
-    assert_equal 3, @game.lowest_total_score
-  end
 #deliverable
   def test_it_can_count_games_by_season
     assert_equal ({20122013=>2, 20162017=>5, 20142015=>6, 20132014=>4, 20152016=>2, 20172018=>1}), Game.count_of_games_by_season
   end
+
+  def test_it_returns_games_goals_by_hoa_team_id
+    game1 = mock
+    game2 = mock
+    game3 = mock
+    game4 = mock
+    game1.stubs(:away_team_id).returns(1)
+    game2.stubs(:away_team_id).returns(1)
+    game3.stubs(:away_team_id).returns(2)
+    game4.stubs(:away_team_id).returns(3)
+    game1.stubs(:home_team_id).returns(11)
+    game2.stubs(:home_team_id).returns(11)
+    game3.stubs(:home_team_id).returns(12)
+    game4.stubs(:home_team_id).returns(13)
+    game1.stubs(:away_goals).returns(1)
+    game2.stubs(:away_goals).returns(10)
+    game3.stubs(:away_goals).returns(2)
+    game4.stubs(:away_goals).returns(3)
+    game1.stubs(:home_goals).returns(5)
+    game2.stubs(:home_goals).returns(10)
+    game3.stubs(:home_goals).returns(7)
+    game4.stubs(:home_goals).returns(8)
+    game_array = [game1,game2,game3,game4]
+    away_hash_hash = Game.hash_of_hashes(game_array, :away_team_id, :goals, :games_played, :away_goals, 1)
+    home_hash_hash = Game.hash_of_hashes(game_array, :home_team_id, :goals, :games_played, :home_goals, 1)
+    Game.stubs(:hash_of_hashes).returns(away_hash_hash)
+    expected = {1 => {:goals => 11, :games_played =>2},
+                2 => {:goals => 2, :games_played =>1},
+                3 => {:goals => 3, :games_played =>1}}
+    assert_equal expected, Game.games_goals_by(:away_team)
+    Game.stubs(:hash_of_hashes).returns(home_hash_hash)
+    expected = {11 => {:goals => 15, :games_played =>2},
+                12 => {:goals => 7, :games_played =>1},
+                13 => {:goals => 8, :games_played =>1}}
+    assert_equal expected, Game.games_goals_by(:home_team)
+  end
```

Comment:
stubs are a fantastic way to get your code lines up...

Reply:
yep yep yep, implementing some now and feeling GREAT